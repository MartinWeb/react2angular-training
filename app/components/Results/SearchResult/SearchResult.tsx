import { Joke } from '../Joke';
import './search-result.scss';

export type SearchResultProps = {
    jokes: ChuckNorris.Joke[];
}

export const SearchResult = ({ jokes }: SearchResultProps) => (
    <div className="search-result">
        <>
            {jokes.map(joke => {
                return <Joke joke={joke} key={joke.id} />
            })}
        </>
    </div>
);
