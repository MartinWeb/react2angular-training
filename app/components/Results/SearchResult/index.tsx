import { react2angular } from 'react2angular';
import { SearchResult, SearchResultProps } from './SearchResult';

const SearchResultComponent = react2angular((props: SearchResultProps) => <SearchResult {...props} />, ['jokes']);

export { SearchResultComponent, SearchResult };
