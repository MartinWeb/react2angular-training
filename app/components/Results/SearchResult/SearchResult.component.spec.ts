import { SearchResultComponent } from "./SearchResult.component";

describe('SearchResultComponent', () => {
    test('should have valid snapshot', () => {
        expect(SearchResultComponent.template).toMatchSnapshot();
    });

    test('should have valid bindings', () => {
        expect(SearchResultComponent.bindings).toMatchSnapshot();
    });
});