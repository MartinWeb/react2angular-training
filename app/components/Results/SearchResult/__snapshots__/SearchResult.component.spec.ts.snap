// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultComponent should have valid bindings 1`] = `
{
  "jokes": "<",
}
`;

exports[`SearchResultComponent should have valid snapshot 1`] = `
"<div class="search-result">
    <joke-component joke="joke" ng-repeat="joke in $ctrl.jokes"></joke-component>
</div>"
`;
